src/main/java/com/google/sps/servlets/CreateCalendarServlet.java

```diff
@@ -58,16 +58,42 @@
   protected void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException {
           
+    UserService userService = UserServiceFactory.getUserService();
+    if(!userService.isUserLoggedIn()) {
+        response.setContentType("application/text");
+        response.getWriter().println("You must be logged in to use the calendar function");
+        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        return;
+    }
+
     Calendar calendarService = Utils.loadCalendarClient();
 
     Entity currentFamilyEntity = null;
     
+    Entity userInfoEntity = Utils.getCurrentUserEntity();
+
+    // If current user is not in a family, they cannot add a member
+    if (userInfoEntity == null) {
+        response.setContentType("application/text");
+        response.getWriter().println("You must belong to a family to use the calendar function");
+        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+    }
```

Comment:
This is also something that seems to be replicated across servlets. Move it to a common Util class.

Reply:
done